File: scala/62.scala
object Solution {
    def uniquePaths(m: Int, n: Int): Int = {
        val dp = Array.ofDim[Int](m, n)
        for (i <- 0 until n) dp(0)(i) = 1
        for (i <- 0 until m) dp(i)(0) = 1
        for (i <- 1 until m)
            for (j <- 1 until n)
                dp(i)(j) = dp(i-1)(j) + dp(i)(j-1)      
        dp(m-1)(n-1)
    }
}

File: leetcodeJava/62.java
package leetcodeJava;

class Solution {
    public int uniquePaths(int m, int n) {
        var map = new int[m][n];
        for (var i = 0; i < m; i++)
            for (var j = 0; j < n; j++) {
                if (i == 0 || j == 0) map[i][j] = 1;
                else map[i][j] = map[i][j-1] + map[i-1][j];
            }
        return map[m-1][n-1];
    }
}
